core/src/main/java/io/questdb/std/GenericLexer.java

```diff
@@ -152,20 +156,48 @@ public int getTokenHi() {
         return _hi;
     }
 
-    public CharSequence getUnparsed() {
-        return unparsed == null ? null : immutableOf(unparsed);
+    public void stash() {
+        int count = 0;
+        while (unparsed.size() > 0) {
+            parkedUnparsed.push(unparsed.pop());
+            parkedPosition.push(unparsedPosition.pop());
+            parkedPosition.push(unparsedPosition.pop());
+            count++;
+        }
+        parkedPosition.push(getPosition());
+        parkedPosition.push(count);
+        // clear next because we create a new parsing context
+        next = null;
+    }
+
+    public void unStash() {
```

Comment:
nit: the method name could be `unstash` to be consistent with `unparse`

Reply:
fixed here: https://github.com/questdb/questdb/pull/2037/commits/c5a5a83266c0d88716b16b5782fb6b9718aeeda4